lib/OpenQA/Utils.pm

```diff
@@ -224,6 +224,27 @@ sub needle_info {
     $needle->{name}      = $name;
     $needle->{distri}    = $distri;
     $needle->{version}   = $version;
+
+    # After change the needle's properties form to hash, support backward compatibility
+    return $needle unless $needle->{properties};
+
+    my @properties = ();
```

Comment:
There is no need to initialize the array with an empty list here, it does nothing.

Reply:
I will fix it.